File: taskflow/models/Todo.ts

interface Todo {
	_id: string;
	user_id?: string;
	userId?: string;
	completed: boolean;
	important: boolean;
	body: string;
	tag?: string;
	tag_color?: string;
	time?: string;
	created_at?: string;
	updated_at?: string;
	completed_at?: string;
	tagColor?: string; // Adding camelCase version for frontend use
	category?: string;
	dueDate?: string;
}


export default Todo;

File: models/todo.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Todo struct {
	ID          primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	UserID      primitive.ObjectID `json:"user_id,omitempty" bson:"user_id,omitempty"`
	Completed   bool               `json:"completed" bson:"completed"`
	Important   bool               `json:"important" bson:"important"`
	Body        string             `json:"body" bson:"body" validate:"required,min=1,max=500"`
	CreatedAt   time.Time          `json:"created_at" bson:"created_at"`
	UpdatedAt   time.Time          `json:"updated_at" bson:"updated_at"`
	CompletedAt *time.Time         `json:"completed_at,omitempty" bson:"completed_at,omitempty"`
}
